ui/analyse/src/plugins/analyse.study.ts

```diff
@@ -2,18 +2,17 @@ import { patch } from '../view/util';
 import makeBoot from '../boot';
 import makeStart from '../start';
 import * as studyDeps from '../study/studyDeps';
-import StrongSocket from 'common/socket';
+import { wsConnect } from 'common/socket';
 
 export { patch };
 
 export const start = makeStart(patch, studyDeps);
 export const boot = makeBoot(start);
 
 export function initModule(cfg: any) {
-  site.socket = new StrongSocket(cfg.socketUrl || '/analysis/socket/v5', cfg.socketVersion, {
+  cfg.socketSend = wsConnect(cfg.socketUrl || '/analysis/socket/v5', cfg.socketVersion, {
     receive: (t: string, d: any) => analyse.socketReceive(t, d),
     ...(cfg.embed ? { params: { flag: 'embed' } } : {}),
-  });
-  cfg.socketSend = site.socket.send;
+  }).send;
```

Comment:
it's here. cfg.socketSend = wsConnect(...).send

many other places as well.

my amateur code archaeologist says our ancestors may have used this to customize send behaviors - where a client of socket modifies their own "send" reference in certain conditions.

Reply:
ah, I see. I think the idea was to only have a dependence to the `send` function instead of the whole websocket